src/macaron/vsa/vsa.py

```diff
@@ -0,0 +1,248 @@
+# Copyright (c) 2024 - 2024, Oracle and/or its affiliates. All rights reserved.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/.
+
+"""VSA schema and generation."""
+
+from __future__ import annotations
+
+import base64
+import json
+from datetime import datetime
+from enum import StrEnum
+from importlib import metadata as importlib_metadata
+from typing import Any, TypedDict
+
+
+class Vsa(TypedDict):
+    """The Macaron Verification Summary Attestation.
+
+    For reference, see:
+    * `SLSA <https://slsa.dev/spec/v1.0/verification_summary>`_.
+    * `in-toto <https://github.com/in-toto/attestation/blob/main/spec/predicates/vsa.md>`_.
+    """
+
+    #: The payload type. Following in-toto, this is always ``application/vnd.in-toto+json``.
+    payloadType: str  # noqa: N815
```

Comment:
Should I move this comment
```
# Disable "N815:mixedCase variable in class scope" to conform with in-toto naming conventions.`
```
to the top of the file, since this applies to a lot of fields in the VSA? or do you think explaining for the first attribute is enough? Let me know.

Reply:
Yes, that's a good idea.

Reply:
See 24896445bedb585bf631f884ae2b8986658e8cc3.